src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -632,13 +635,21 @@ public static string GetClassNameFromInstance (IntPtr jobject)
 		}
 
 		[DllImport ("__Internal", CallingConvention = CallingConvention.Cdecl)]
-		internal static extern IntPtr monodroid_typemap_managed_to_java (string managed);
+		static extern IntPtr monodroid_typemap_managed_to_java (byte[] mvid, int token);
+
+		internal static IntPtr TypemapManagedToJava (Type type)
+		{
+			Span<byte> mvid = new Span<byte>(mvid_bytes);
+			type.Module.ModuleVersionId.TryWriteBytes (mvid);
```

Comment:
Should this check the return value of `TryWriteBytes`? And I guess `throw`?

Reply:
Ordinarily, yes, in this case I don't think it's necessary. Memory for the array is preallocated at class instantiation and we know that it's the right size, I don't think we'd see problems here and I want to squeeze every little ounce of performance out of this.

Reply:
I rather doubt that the return code check + exception throw would actually be noticeable...

Reply:
Probably not, but if it's not necessary (we control the whole code), then why waste the ns?